src/modules/string/String.cpp

```diff
@@ -32,6 +37,34 @@ const std::map<HexPrefix, std::string> hexPrefixToStringMapping{
 };
 }
 
+bool isValidGuarantee(GuaranteeMap& guarantee, std::set<char>& targetCharacters, unsigned int length)
+{
+    unsigned int atleastCountSum{};
+    unsigned int atmostCountSum{};
+    for (auto& it : guarantee)
+    {
+        // if a char in guarantee is not in char set, it is an invalid guarantee
+        if (targetCharacters.find(it.first) == targetCharacters.end())
+            return false;
+        atleastCountSum += it.second.atleastCount;
+        atmostCountSum += it.second.atmostCount;
+    }
+    // if atleastCount sums up greater than total length of string, it is an invalid guarantee
+    // if all chars in targetCharacters are mapped in guarantee, we need to check for validity of atmostCount
+    // if atmostCount sumps up less than total length of string, it in an invalid guarantee
+    if (atleastCountSum > length || (guarantee.size() == targetCharacters.size() && atmostCountSum < length))
+        return false;
+    return true;
+}
+
+std::string generateAtleastString(const GuaranteeMap& guarantee)
+{
+    std::string result;
+    for (auto& it : guarantee)
+        result += std::string(it.second.atleastCount, it.first);
```

Comment:
Could you please use curly braces even for one-liner? We have convention like that in the codebase. It applies to all occurences in changes like aboe for if statement etc